src/azul/collections.py

```diff
@@ -62,35 +64,44 @@ def explode_dict(d: Mapping[K, Union[V, List[V], Set[V], Tuple[V]]]) -> Iterable
         yield dict(zip(d.keys(), t))
 
 
-def none_safe_key(v):
+def none_safe_key(none_last: bool = True) -> Callable[[Any], Any]:
     """
-    A sort key that handles None values.
+    A sort key factory that handles None values.
 
     >>> sorted([1, None, 2])
     Traceback (most recent call last):
     ...
     TypeError: '<' not supported between instances of 'NoneType' and 'int'
 
-    >>> sorted([1, None, 2], key=none_safe_key)
+    >>> sorted([1, None, 2], key=none_safe_key())
+    [1, 2, None]
+
+    >>> sorted([1, None, 2], key=none_safe_key(none_last=False))
     [None, 1, 2]
     """
-    return v is not None, v
+    if none_last:
+        return lambda v: (v is None, v)
+    else:
+        return lambda v: (v is not None, v)
 
 
-def none_safe_tuple_key(t):
+def none_safe_tuple_key(none_last: bool = True) -> Callable[[Any], Any]:
     """
-    A sort key that handles tuples containing None value.
+    A sort key factory that handles tuples containing None value.
 
     >>> sorted([(2, 'c'), (None, 'a'), (2, None), (1, 'b')])
     Traceback (most recent call last):
     ...
     TypeError: '<' not supported between instances of 'NoneType' and 'int'
 
-    >>> sorted([(2, 'c'), (None, 'a'), (2, None), (1, 'b')], key=none_safe_tuple_key)
+    >>> sorted([(2, 'c'), (None, 'a'), (2, None), (1, 'b')], key=none_safe_tuple_key())
+    [(1, 'b'), (2, 'c'), (2, None), (None, 'a')]
+
+    >>> sorted([(2, 'c'), (None, 'a'), (2, None), (1, 'b')], key=none_safe_tuple_key(none_last=False))
     [(None, 'a'), (1, 'b'), (2, None), (2, 'c')]
     """
-    assert isinstance(t, tuple)
-    return tuple(map(none_safe_key, t))
+    # assert isinstance(t, tuple)
```

Comment:
The assert should stay. It just needs to move into the body of the returned callable.